src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Tooltip/TagHelperTooltipFactoryBase.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Razor.Tooltip;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Tooltip
+{
+    internal abstract class TagHelperTooltipFactoryBase<T>
```

Comment:
Everything in this class was moved from the old DefaultTagHelperTooltipFactory class. Contains the mutual logic between `VSLSPTagHelperTooltipFactory` and `LSPTagHelperTooltipFactory`.